storage/src/node/branch.rs

```diff
@@ -324,6 +324,18 @@ impl BranchNode {
                 Some(Child::AddressWithHash(_, hash)) => Some((i, hash)),
             })
     }
+
+    /// Returns (index, address) for each child that has a hash set.
+    pub fn children_addresses(&self) -> impl Iterator<Item = (usize, &LinearAddress)> + Clone {
```

Comment:
These are almost exactly the same, so let's create a helper that returns `(i, Child::AddressWithHash)`